apps/server/src/modules/oauth/service/iserv-oauth.service.ts

```diff
@@ -0,0 +1,29 @@
+import { Injectable } from '@nestjs/common';
+import { User } from '@shared/domain';
+import { UserRepo } from '@shared/repo';
+import { Logger } from '@src/core/logger';
+import { OAuthSSOError } from '../error/oauth-sso.error';
+import { IJWT } from '../interface/jwt.base.interface';
+
+@Injectable()
+export class IservOAuthService {
+	constructor(private readonly userRepo: UserRepo, private logger: Logger) {
+		this.logger.setContext(IservOAuthService.name);
+	}
+
+	extractUUID(decodedJwt: IJWT): string {
+		if (!decodedJwt || !decodedJwt.uuid) throw new OAuthSSOError('Failed to extract uuid', 'sso_jwt_problem');
+		const { uuid } = decodedJwt;
+		return uuid;
+	}
+
+	async findUserById(uuid: string, systemId: string): Promise<User> {
+		let user: User;
+		try {
+			user = await this.userRepo.findByLdapId(uuid, systemId);
+		} catch (error) {
```

Comment:
only throw when user really was not found. why does findByLdapId throws an error and does not return null?

Reply:
Please dont throw an http not found in the user repo

Reply:
we are using a promise.reject now, it will return undefined